internal/packages/walk.go

```diff
@@ -432,27 +461,16 @@ func (r *symlinkResolver) follow(dir, base string) bool {
 		// Bazel to point to internal build directories.
 		return false
 	}
-	// See if the symlink points to a tree that has been already visited.
-	fullpath := filepath.Join(dir, base)
-	dest, err := filepath.EvalSymlinks(fullpath)
+	canonical, err := filepath.EvalSymlinks(filepath.Join(dir, base))
 	if err != nil {
 		return false
 	}
-	if !filepath.IsAbs(dest) {
-		dest, err = filepath.Abs(filepath.Join(dir, dest))
-		if err != nil {
-			return false
-		}
-	}
-	for _, p := range r.visited {
```

Comment:
I don't think I actually understand what this is doing. I replaced it with a check whether the symlink is inside the repository. I think that will work for your use case, but please confirm.

Reply:
This is to handle the case where we have two symlinks, one pointing to, e.g., /external/foo and another pointing to /external/foo/bar. This is just a paranoid check, so I'm fine with removing it.